csv_joiner.py

```diff
@@ -0,0 +1,147 @@
+import sys
+
+
+class InvalidModeError(Exception):
+    def __init__(self, modes):
+        self.expected = '/'.join(modes)
+        self.message = 'Invalid join mode'
+
+    def __str__(self):
+        return f'{self.message}. Expected: {self.expected}'
+
+
+class InvalidColumnError(Exception):
+    def __init__(self, col):
+        self.col = col
+        self.message = 'Column "{0}" does not appear in one/both files'
+
+    def __str__(self):
+        return self.message.format(self.col)
+
+
+class Parser:
+    @staticmethod
+    def parse_line(line: str) -> list:
+        line = line.rstrip()
+        start_i = 0
+        items = []
+        quotes = 0
+        for i, char in enumerate(line):
+            if char == ',' and not quotes:
+                items.append(line[start_i:i])
+                start_i = i + 1
+
+            if char == '"':
+                quotes += 1
+
+            try:
+                if line[i + 1] == ',' and quotes % 2 == 0:
+                    quotes = 0
+            except IndexError:
+                break
+
+        items.append(line[start_i:])
+        return items
+
+
+class Line:
+    @staticmethod
+    def add_to_dict(list_of_items: list, keys_dict: dict):
+        new_dict = {}
+        for item, key in zip(list_of_items, keys_dict.keys()):
+            new_dict[key] = item
+        return new_dict
+
+    @staticmethod
+    def print_line(merged_dict: dict, order: list):
+        out_line = ''
+        for col in order:
+            out_line += ',' + merged_dict[col]
+        print(out_line[1:])
+
+
+class Header:
+    order = []
+
+    def __init__(self, line):
+        self.line = line
+        self.list = Parser.parse_line(self.line)
+        self.dict = self.list_to_dict()
+
+    def list_to_dict(self):
+        headers_dict = {}
+        for i in self.list:
+            headers_dict[i] = 'None'
+        return headers_dict
+
+    @staticmethod
+    def merge_headers_keys(h1, h2) -> dict:
+        merged_headers = h1.dict
+        merged_headers.update(h2.dict)
+        return merged_headers
+
+    @staticmethod
+    def set_order(h1, h2, key_col):
+        if key_col not in h1.list or key_col not in h2.list:
+            raise InvalidColumnError(key_col)
+
+        headers_order = [key_col]
+        headers_order += [h for h in h1.list if h not in headers_order]
+        headers_order += [h for h in h2.list if h not in headers_order]
+        return headers_order
```

Comment:
Arguably, this doesn't set the order, rather it sorts the columns ;)
Is this order correct? Is it standard? Why this order, not o=any other?

Reply:
I decided on this order to make output data more organized: first column is the chosen one which joins the given two files and then all columns from left file and at the end all columns from the second file. This order isn't standard. For now I would rather keep it simple like it's in sql.